camel-kafka-connector-catalog/src/main/java/org/apache/camel/kafkaconnector/catalog/CamelKafkaConnectorCatalog.java

```diff
@@ -0,0 +1,138 @@
+                    connectorsName.add("camel-coap+tcp-sink");
+                } else {
+                    connectorsName.add(connector);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            LOG.error("Cannot find file: {}", e.getMessage(), e);
+        } catch (IOException e) {
+            LOG.error("IO Exception: {}", e.getMessage(), e);
+        }
+    }
+
+    public String getConnectorAsJson(String connectorName) {
+        String result = null;
+        try {
+            result = Files.lines(Paths.get(CONNECTORS_DIR + File.separator + connectorName + ".json")).parallel() // for
+                                                                                                                        // parallel
+                                                                                                                        // processing
+                .map(String::trim) // to change line
+                .collect(Collectors.joining());
+        } catch (IOException e) {
+            LOG.error("IO Exception: {}", e.getMessage(), e);
+        }
+        return result;
+    }
+
+    private CamelKafkaConnectorModel getConnectorModel(String connectorName) {
+        CamelKafkaConnectorModel model = new CamelKafkaConnectorModel();
+        String json = getConnectorAsJson(connectorName);
+        JsonObject obj = JsonMapper.deserialize(json);
+        JsonObject wrapper = (JsonObject)obj.get("connector");
+        model.setConnectorClass((String)wrapper.get("class"));
+        model.setArtifactId((String)wrapper.get("artifactId"));
+        model.setGroupId((String)wrapper.get("groupId"));
+        model.setType((String)wrapper.get("type"));
+        model.setVersion((String)wrapper.get("version"));
+        model.setOptions((List<CamelKafkaConnectorOptionModel>)getConnectorOptionModel(obj));
+        return model;
+    }
+
+    private List<CamelKafkaConnectorOptionModel> getConnectorOptionModel(JsonObject obj) {
+        List<CamelKafkaConnectorOptionModel> model = new ArrayList<CamelKafkaConnectorOptionModel>();
+        JsonObject wrapper = (JsonObject)obj.get("properties");
+        Set<String> options = wrapper.keySet();
+        for (String string : options) {
+            JsonObject object = (JsonObject)wrapper.get(string);
+            CamelKafkaConnectorOptionModel singleModel = new CamelKafkaConnectorOptionModel();
+            singleModel.setDefaultValue((String)object.get("defaultValue"));
+            singleModel.setPriority((String)object.get("priority"));
+            singleModel.setDescription((String)object.get("description"));
+            singleModel.setName((String)object.get("name"));
+            model.add(singleModel);
+        }
+        return model;
+    }
+
+    public List<String> getConnectorsName() {
+        return connectorsName;
+    }
+
+    public static Map<String, CamelKafkaConnectorModel> getConnectorsModel() {
```

Comment:
why this method is static?

If accessed before an instanciation of the class happened, it will return an empty value.

Reply:
It's a mistake.

Reply:
ok, I will provide a PR to remove the static

Reply:
It's already done and merged